src/irmin-pack/unix/file_manager.ml

```diff
@@ -255,6 +255,14 @@ struct
     t.suffix <- suffix1;
     Suffix.close suffix0
 
+  let reload_lower t =
```

Comment:
Instead of scattering comments, I'll co-locate some thoughts about overall architecture here since it is the first function in the diff.

First some background/context. The two primary function on file manager that deal with post-GC finalization are `swap` and `reload`:
- `swap` is called by `gc.ml` as a part of finalization in its `swap_and_purge` function. This is, necessarily, called in the RW instance.
- `reload` is called by RO instances to update their view of the world to the latest. For octez, it is called after every commit, so performance considerations are important.

Given this, I would like to propose:
- Remove `reload_lower` as a public API for file manager. Lets maintain just `swap` and `reload` as the public functions.
- Keep it as an internal function that can be used by `swap` and `reload`
- Move `swap_volume` from `gc.ml` to `swap` in file manager. I think it will be nicer to have all the "swap" logic in one place, and be in line with file manager's file managing role.
- For `reload_lower`, it should take into consideration volume number changes since the RO could be out of date.
- When calling `reload_lower` from `reload`, lets only do it when necessary. I think the criteria should be if generation number changes or volume number changes. We already have some logic in there for handling generation number changes.

WDYT?

Reply:
"That looks reasonable." :p

Thank you for the context on the architecture! I was pretty sure this code wasn't in the right place but I wanted to get something working before anything else

Reply:
This has been implemented and seems to work!